tests/sources/test_network_drive.py

```diff
@@ -969,3 +985,26 @@ async def test_list_file_permissions_with_inaccessible_file():
             )
 
             assert result is None
+
+
+@pytest.mark.asyncio
+@mock.patch.object(
+    NASDataSource,
+    "list_file_permission",
+    return_value=[
+        mock_permission(sid="S-2-21-211-411", ace=0),
+        mock_permission(sid="S-1-11-10", ace=1),
```

Comment:
If you add a user here that _does_ have access and does belong to the group, in addition to there being one user who was explicitly denied, what happens?

I expect that the document ends up with the group being granted allow permissions. Right?

Reply:
Hey, In that case the second user _won't_ be able to access the document since group permissions are also counted as explicit ACEs. Now Explicit Deny (from group membership) has precedence over Explicit allow (individual), so document is not accessible. 

I'll update the test case to add the second user to the group highlighting this scenario.

Reply:
I thought this case was testing a group with ALLOW access, but a single user with explicit DENY access?

What I'd like to validate is the scenario where:

```
Users
-----
ID | First | Last
1 | John | Doe
2 | Jane | Fonda

Groups
-------
ID | Users
1 | [1, 2]
```

where Group 1 is given ALLOW access, User 1 is given explicit ALLOW access, and User 2 is given explicit DENY access. What is the resulting document `_allow_access_control` and the `permissions` for each of the two users?

Reply:
In that scenario the resulting document's `_allow_access_control` is [`id-user1`], and the document will be accessible to User 1, and inaccessible to User 2. Should I add another test case it ?

Reply:
Please do - because reading the code, I'm worried the document will end up with `_allow_access_control` of `[group1]`, which would end up giving both users access, in error

Reply:
Thanks - I'm glad to see I was wrong, and I'm glad we'll have this test to protect against regressions.